### PR TITLE
Allow for installation of Powercord plugins

### PR DESCRIPTION
**Pull request by vbajs**
This pull request adds the ability for addon installer to install powercord plugins only under 2 conditions:
1. The user is currently in the plugin-links channel of powercord's server
2. The user has [pccompat](https://github.com/MoltenCoreDev/pcCompat) ~both installed and~ enabled

~It works perfectly but 2 problems remain~

~1. If the plugin entry has multiple links, then it cannot install the powercord plugin even when the user press the "Install Plugin" button (no errors are thrown)~
~2. If the plugin developer removed the embed of their plugin link in their entry, then an error is thrown in devtools console~

~I am certain these 2 problems could both be soon fixed but for as long as I have been working on this, I'm glad that the essentials functions work~
All fixed as per cb146403589907f511c3a76f454f751254348ef5 and dfcf88360c162caccf6fd01b8b9516e6780c3e79, thanks to @LavaSquids for providing a working regex!
Feel free to review this pull request!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,49 +9,80 @@ export default class AddonInstaller extends Plugin {
   start () {
     patch(
       'addon-installer',
-      getModule((m) => m.default?.displayName === 'MessageContextMenu'),
+      getModule((m) => m.default && m.default?.displayName === 'MessageContextMenu'),
```
Comment (valentinegb) [inline index.js]: Unnecessary change, redundant. `m.default?.displayName` is exactly the same as `m.default && m.default.displayName`.

**Review by valentinegb — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (vbajs) [reply index.js]: Done as per d09ed48255772fa1fccbd85f867d34fa2a762fa1

**Review by valentinegb — DISMISSED** [review]

File: index.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ import { patch, unpatchAll } from '@vizality/patcher';
 import { getModule } from '@vizality/webpack';
 import { ContextMenu } from '@vizality/components';
 
+const githubRegex = /https?:\/\/github\.com\/[a-zA-Z0-9-]+\/[a-zA-Z0-9-]+\/?$/m;
```
Comment (valentinegb) [inline index.js]: Only referenced once, does not need to be defined as a constant.

File: index.js [hunk 3]
```diff
@@ -13,45 +15,76 @@ export default class AddonInstaller extends Plugin {
...
-                />
-              </ContextMenu.Group>
-            </>
-          );
-        }
+        if (channelID === '755005584322854972') {
+          if (vizality.manager.plugins.isInstalled('00pccompat') && vizality.manager.plugins.isEnabled('00pccompat')) {
+            const addonURL = args[0].message?.content.match(githubRegex)[0];
+            const addonID = addonURL.split('/').pop().toLowerCase();
+            const isPlugin = channelID === '755005584322854972' && true;
```
Comment (valentinegb) [inline index.js]: Variables with the same name defined twice. Consider defining at a higher level and making it work for both scenarios.

File: index.js [hunk 4]
```diff
@@ -13,45 +15,76 @@ export default class AddonInstaller extends Plugin {
...
-                      await vizality.manager[pluginsOrThemes].install(addonURL);
-                    }
-                  }}
-                />
-              </ContextMenu.Group>
-            </>
-          );
-        }
+        if (channelID === '755005584322854972') {
+          if (vizality.manager.plugins.isInstalled('00pccompat') && vizality.manager.plugins.isEnabled('00pccompat')) {
```
Comment (valentinegb) [inline index.js]: The plugin cannot be enabled without also being installed, it is only necessary to check that it is enabled.

File: index.js (hunk 3 above)
Comment (valentinegb) [reply index.js]: Additionally, the `&& true` is unnecessary. You can achieve the same result with `const isPlugin = channelID === '755005584322854972';`.

Comment (valentinegb) [reply index.js]: In this specific scenario, `isPlugin` will always be true because it is inside an if statement checking that `channelID === '755005584322854972'` and is therefore entirely unnecessary.

File: index.js [hunk 5]
```diff
@@ -13,45 +15,76 @@ export default class AddonInstaller extends Plugin {
...
+            const addonURL = args[0].message?.content.match(githubRegex)[0];
+            const addonID = addonURL.split('/').pop().toLowerCase();
+            const isPlugin = channelID === '755005584322854972' && true;
+            const addonIsInstalled = vizality.manager.plugins.keys.includes(addonID);
+
+            if (isPlugin) {
```
Comment (valentinegb) [inline index.js]: Will always be true, can be removed.

**Review by valentinegb — CHANGES_REQUESTED** [review]
I didn't review every line of code, after the first 6 comments, I realized there was ***a lot*** of redundancy and unnecessary code. I hope you can fix the issues I pointed out and the ones I didn't.

File: index.js (hunk 2 above)
Comment (vbajs) [reply index.js]: I suppose as a defined as variable is more suitable?

Comment (vbajs) [reply index.js]: Also esinit prefers it to be defined as a constant

File: index.js (hunk 3 above)
Comment (vbajs) [reply index.js]: Done as per 4f1b7eb22b6c6933af4d221f82d970bc5a645c9d

File: index.js (hunk 4 above)
Comment (vbajs) [reply index.js]: Done as per 4f1b7eb22b6c6933af4d221f82d970bc5a645c9d

File: index.js (hunk 5 above)
Comment (vbajs) [reply index.js]: Done as per 4f1b7eb22b6c6933af4d221f82d970bc5a645c9d

File: index.js (hunk 2 above)
Comment (vbajs) [reply index.js]: Done as per 8387a2bed1d443973701fb2e7406733d8e08dfd8

File: index.js [hunk 6]
```diff
@@ -13,45 +13,72 @@ export default class AddonInstaller extends Plugin {
...
+                        await vizality.manager[pluginsOrThemes].uninstall(
+                          addonID
+                        );
+                      } else {
+                        await vizality.manager[pluginsOrThemes].install(addonURL);
+                      }
+                    }}
+                  />
+                </ContextMenu.Group>
+              </>
```
Comment (valentinegb) [inline index.js:75]: This context menu group is *veeeery* similar to the one for Powercord. Would it be possible to merge them?

**Review by valentinegb — CHANGES_REQUESTED** [review]

File: index.js (hunk 6 above)
Comment (vbajs) [reply index.js:75]: Not really, we can only have powercord plugins support, but not themes support without the plugin somehow renaming `powercord_manifest` to `manifest`
But if you mean to make it outside of the if statement and return to it when it has everything, then I'll try that

**Comment by valentinegb** [comment]
This feature has been implemented as of commit 3796075.
